File: pset6/cash.py
# Imports libraries.
import cs50
import time

# Initializes constants.
QUARTERS = .25
DIMES = .10
NICKELS = .05
PENNIES = .01

# Gets Input.
change = cs50.get_float("Change owed: ")

# Tests, and if needed, rerequests input.
while change < 0:
    change = cs50.get_float("Change Owed: ")

# Intializes counter.
counter = 0

# Counts quarters.
while change >= QUARTERS:
    change = change - QUARTERS
    counter += 1

# Rounds change to clear junk values.
change = round(change, 2)

# Counts dimes.
while change >= DIMES:
    change = change - DIMES
    counter += 1

# Rounds change to clear junk values.
change = round(change, 2)

# Counts nickels.
while change >= NICKELS:
    change = change - NICKELS
    counter += 1

# Rounds change to clear junk values.
change = round(change, 2)

# Counts pennies.
while change >= PENNIES:
    change = change - PENNIES
    counter += 1

# Returns total coin amount.
print(counter)

File: pset0/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main()
{
// Initializing Constants
    float QUARTERS = 25;
    float DIMES = 10;
    float NICKELS = 05;
    float PENNIES = 01;

// Gets Change Input, and rerequests if necessary.
    int coins = 0;
    double money = get_float("Change owed? ");

// Tests Change input, and rerequests if necessary.
    while (money < 0)
    {
        money = get_float("Change owed? ");

    }

// Rounds money value to begin operation.
    money *= 100;

    money = round(money);

// Checks maximum amount of quarters usable.
    while (money >= QUARTERS)
    {
        money -= QUARTERS;
        coins += 1;
    }

// Checks maximum amount of dimes usable after quarters.
    while (money >= DIMES)
    {
        money -= DIMES;
        coins += 1;
    }

// Checks maximum amount of nickels usable after dimes.
    while (money >= NICKELS)
    {
        money -= NICKELS;
        coins += 1;
    }

// Checks maximum amount of pennies usable after nickels.
    while (money >= PENNIES)
    {
        money -= PENNIES;
        coins += 1;
    }

// Returns output.
    printf("%i\n", coins);







}